telemetry/telemetrybuffer.go

```diff
@@ -319,14 +326,15 @@ func (tb *TelemetryBuffer) ConnectToTelemetryService(telemetryNumRetries, teleme
 	}
 }
 
-func getTelemetryServiceDirectory() (path string, dir string) {
+func getTelemetryServiceDirectory() (path string, dir string, fileExists bool) {
```

Comment:
I think it should be the responsibility of the caller to check if the returned path exists, and this method should only be responsible for composing the path from the config

Reply:
Talked to Evan and implemented agnostic calls for "Sprintf" and "platform". Changed the implementation to reflect method naming to prevent over responsibility.